doc/tor.md

```diff
@@ -90,41 +114,110 @@ and open port 8333 on your firewall (or use -upnp).
+onion nodes.
+
+This feature is enabled by default if Bitcoin Core is listening (`-listen`) and
+it requires a Tor connection to work. It can be explicitly disabled with
+`-listenonion=0`. If it is not disabled, it can be configured using the
+`-torcontrol` and `-torpassword` settings.
+
+To see verbose Tor information in the bitcoind debug log, pass `-debug=tor`.
+
+### Control Port
+
+You may need to set up the Tor Control Port. On most Linux distributions there
+will be the following commented-out settings in `/etc/tor/torrc`:
+
+```
+ControlPort 9051
+CookieAuthentication 1
+CookieAuthFileGroupReadable 1
+```
+
+Uncomment those, save, and restart Tor (usually `systemctl restart tor` or `sudo
+systemctl restart tor` on most systemd-based systems, including recent Debian
+and Ubuntu, or just restart the computer).
+
+On some systems (such as Arch Linux), you may also need to add the following
+line:
+
+```
+DataDirectoryGroupReadable 1
+```
+
+### Authentication
+
+Connecting to Tor's control socket API requires one of two authentication
+methods to be configured: cookie authentication or bitcoind's `-torpassword`
+configuration option.
+
+#### Cookie authentication
+
+For cookie authentication, the user running bitcoind must have read access to
+the `CookieAuthFile` specified in the Tor configuration. In some cases this is
+preconfigured and the creation of an onion service is automatic. Don't forget to
+use the `-debug=tor` bitcoind configuration option to enable Tor debug logging.
+
+If a permissions problem is seen in the debug log, e.g. `tor: Authentication
+cookie /run/tor/control.authcookie could not be opened (check permissions)`, it
+can be resolved by adding both the user running Tor and the user running
+bitcoind to the same Tor group and setting permissions appropriately.
+
+To know the name of the Tor group, list all groups and grep for a "tor" group
+name:
+
+```
+getent group | cut -d: -f1 | sort | grep -i tor
+```
+
+On Debian-derived systems, the Tor group will most likely be `debian-tor`.
+
+You can also check the group of the cookie file. On most Linux systems, the Tor
+auth cookie will usually be `/run/tor/control.authcookie`:
+
+```
+stat -c '%G' /run/tor/control.authcookie
+```
+
+Once you have determined the `${TORGROUP}` and selected the `${USER}` that will
+run bitcoind, run this as root:
+
+```
+usermod -a -G ${TORGROUP} ${USER}
+```
+
+Then restart the computer (logging out and back in again should also work), and
+confirm that the user is in the Tor group by running the groups command above.
+
+If the `/run/tor/control.authcookie` exists in your system, log in as the user
+that will run bitcoind and run this command:
+
+```
+cat /run/tor/control.authcookie > /dev/null
+```
+
+If the above prints nothing and returns, Bitcoin Core should work with your Tor
+configuration. If it prints an error, a configuration problem will likely
+prevent Bitcoin Core from working with your Tor.
```

Comment:
I think this is excessive and should be removed.

* the "sort" in the getent command is unneeded and on my computer the entire thing prints:

```
operator
_tor
You have new mail in /home/toolame/mailbox
```

so, if I am a dummy user, how do I know which is the tor group?

* stat gives me error:
```
$ stat -c '%G' /run/tor/control.authcookie
stat: illegal option -- c
usage: stat [-FLnq] [-f format | -l | -r | -s | -x] [-t timefmt] [file|handle ...]
```

This doc shouldn't be targeted for users who lack basic knowledge of how filesystem permissions work.

Reply:
Are you using a Debian-based system?

Reply:
Dropped the commit.

Reply:
Revived without the `sort` and with a mention that the command is for Debian in #20757.